spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -73,16 +75,35 @@ public class Mqttv5PahoMessageHandler extends AbstractMqttMessageHandler
 
 	public Mqttv5PahoMessageHandler(String url, String clientId) {
 		super(url, clientId);
-		this.connectionOptions = new MqttConnectionOptions();
-		this.connectionOptions.setServerURIs(new String[]{ url });
-		this.connectionOptions.setAutomaticReconnect(true);
+		this.connectionOptions = buildDefaultConnectionOptions(url);
 	}
 
 	public Mqttv5PahoMessageHandler(MqttConnectionOptions connectionOptions, String clientId) {
 		super(obtainServerUrlFromOptions(connectionOptions), clientId);
 		this.connectionOptions = connectionOptions;
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @since 6.0
+	 */
+	public Mqttv5PahoMessageHandler(ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager) {
+		super(clientManager);
+		this.connectionOptions = clientManager.getConnectionInfo();
+	}
+
+	private static MqttConnectionOptions buildDefaultConnectionOptions(@Nullable String url) {
+		final MqttConnectionOptions connectionOptions;
+		connectionOptions = new MqttConnectionOptions();
+		if (url != null) {
+			connectionOptions.setServerURIs(new String[]{ url });
+		}
+		connectionOptions.setAutomaticReconnect(true);
```

Comment:
No, we don't need this here: if client is not connect on send, we try to connect it first.

Reply:
But it was used in the existing implementation, see 78 line in the existing impl.